src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509Certificate2.cs

```diff
@@ -778,6 +779,116 @@ public X509Certificate2 CopyWithPrivateKey(ECDiffieHellman privateKey)
             return new X509Certificate2(pal);
         }
 
+        /// <summary>
+        ///   Get the <see cref="MLDsa"/> public key from this certificate.
+        /// </summary>
+        /// <returns>
+        ///   The public key, or <see langword="null"/> if this certificate does not have an ML-DSA public key.
+        /// </returns>
+        /// <exception cref="PlatformNotSupportedException">
+        ///   The certificate has an ML-DSA public key, but the platform does not support ML-DSA.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   The public key was invalid, or otherwise could not be imported.
+        /// </exception>
+        [Experimental(Experimentals.PostQuantumCryptographyDiagId)]
+        public MLDsa? GetMLDsaPublicKey()
+        {
+            MLDsaAlgorithm? algorithm = MLDsaAlgorithm.GetMLDsaAlgorithmFromOid(GetKeyAlgorithm());
+
+            if (algorithm is null)
+            {
+                return null;
+            }
+
+            byte[] publicKey = Pal.PublicKeyValue;
+
+            if (publicKey.Length != algorithm.PublicKeySizeInBytes)
```

Comment:
why do we need to check the size? (or why not in GetMLDsaPrivateKey)

Reply:
ImportMLDsaPublicKey throws an ArgumentException for the same case.  Though looks like I never came back and updated this to check that the parameters was empty, so this should probably just export SPKI and call the import SPKI.

GetMLDsaPrivateKey didn't do this check because it's not parsing data, it just has the key handle.